### ISSUE TITLE
NeurlPS '19 | XLNet: Generalized Autoregressive Pretraining for Language Understanding

### ISSUE THREAD

**Issue by vycezhong** [vycezhong/read-papers#138]
https://arxiv.org/pdf/1906.08237.pdf

**Comment by vycezhong** [comment]
粗浅看了下. 上来先批判了autoregressive(AR)和autoencoding(AE)这两种主流无监督表示学习方法. AR无法学到双向信息，AE引入[MASK]带来pre-train-finetune discrepancy. 

XLNet引入了一个新的task - PLM (permutation language modeling). AR是left-to-right或者right-to-left将句子分解，即P(x) = p(x1)p(x2|x1)p(x3|x1,x2) ...  而XLNet会乱序分解句子，P(x) = p(x3)p(x2|x3)p(x4|x2,x3)p(x1|x2,x3,x4). 这样可以x3也利用前后x2, x4的信息.  具体实现并不是真的打乱词序，而是用合适的attention mask来实现这一点. 另外，XLNet采用了transformer-xl作为backbone, 这应该是xlnet名字由来. 

pretraining上，选用了更大的dataset，大概是bert的10倍，在512TPUv3上训练了500K steps 2.5 days. optimizer还是Adam. bs = 2048. 这是XLNet-large. 还训练了XLNet-base，使用了和bert相同的dataset. 

SQuAD1.1
![image](https://user-images.githubusercontent.com/25879526/103619285-6be27e80-4f6c-11eb-8b3f-b5ffd7445b05.png)

大幅涨点... 不过这个比较不是很公平. 

XLNet-Base这个比较更为公平. 涨点还是很吓人. 去掉了memory(transformer-xl)后，还是很高. 说这个比BERT更nb应该没啥问题...有点惊讶的是NSP和PLM不太兼容. 用了NSP反而掉分. 
![image](https://user-images.githubusercontent.com/25879526/103619418-a815df00-4f6c-11eb-9f65-9095e9706618.png)
